src/Nethermind/Nethermind.Mev/MevPlugin.cs

```diff
@@ -112,13 +112,23 @@ public Task InitRpcModules()
                 IJsonRpcConfig rpcConfig = getFromApi.Config<IJsonRpcConfig>();
                 rpcConfig.EnableModules(ModuleType.Mev);
 
+                BlockValidationService blockValidationService = new(getFromApi.MainBlockProcessor!,
+                    getFromApi.TransactionProcessor!,
+                    getFromApi.StateProvider!,
```

Comment:
You cannot use main state and main transaction processor here. You need to use ReadOnlyTxProcessingEnvironment like with any other RPC.